wsmaster/codenvy-hosted-api-license/src/main/java/com/codenvy/api/license/server/LicenseService.java

```diff
@@ -199,4 +212,33 @@ public Response isCodenvyUsageLegal() throws ApiException {
         }
     }
 
+    @GET
+    @Path("/legality/node")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Check if Codenvy node usage matches Codenvy License constraints, or free usage rules. If parameter nodeNumber " +
+                          "is required, then it will be validated, otherwise number swarm nodes will be validated")
```

Comment:
In fact, it is not always optional. It could have a value in case of requesting from *Add Node* IM CLI command.